presto-main/src/main/java/com/facebook/presto/cost/AggregationStatsRule.java

```diff
@@ -76,27 +95,50 @@ public static PlanNodeStatsEstimate groupBy(PlanNodeStatsEstimate sourceStats, C
             }));
         }
 
-        double rowsCount = 1;
+        double rowsCount = SINGLE_ROW_COUNT;
         for (VariableReferenceExpression groupByVariable : groupByVariables) {
             VariableStatsEstimate symbolStatistics = sourceStats.getVariableStatistics(groupByVariable);
             int nullRow = (symbolStatistics.getNullsFraction() == 0.0) ? 0 : 1;
             rowsCount *= symbolStatistics.getDistinctValuesCount() + nullRow;
         }
-        result.setOutputRowCount(min(rowsCount, sourceStats.getOutputRowCount()));
+        // groupBy key may be a derived expression which may not have associated stats hence return NaN.
+        // In such case, use source stats to bound the output row count
+        if (!isNaN(rowsCount)) {
+            result.setOutputRowCount(min(rowsCount, sourceStats.getOutputRowCount()));
+        }
+        else {
+            result.setOutputRowCount(sourceStats.getOutputRowCount());
```

Comment:
This assume the input and output count is the same when no group by key stats available? Just curious if there is any reason/benefit of doing this?

Reply:
Thanks for reviewing the code. The assumption is that if the agg columns cannot guaranteed uniqueness, the agg estimates takes the source output row count as the upper bound limit to avoid returning NaN which breaks the join reordering.